bots/mainbot/castle.js

```diff
@@ -1,172 +1,352 @@
 'use strict';
 
-import {BCAbstractRobot, SPECS} from 'battlecode';
+import { BCAbstractRobot, SPECS} from 'battlecode';
 import nav from './nav.js' ;
 import utilities from './utilities.js'
 import combat from './combat.js'
 
 const castle = {};
 const pilgrims ={};
-
-
 pilgrims.number = 0;
-
-var first_castle = true ;
-var castle_locs = [];
-
-
+var castle_ids = [];
+var builds = 0;
+var buildunitflag =0 ;
 castle.takeTurn = (self) => {
-	self.log('castle taking turn')
 	
+	self.log('castle taking turn')
 	var robotsnearme = self.getVisibleRobots();
 
-	var getBuildDir = function(buildunit) {
-        var options = nav.rotate_arr.filter((d) => {
-            return nav.isPassable(nav.applyDir(self.me, d), self.getPassableMap(), self.getVisibleRobotMap())
-        })
-        return options[0];
-	}
-
+			
 	var attackable = robotsnearme.filter((r) => {
-        if (! self.isVisible(r)){
-            return false;
-        }
-        const dist = (r.x-self.me.x)**2 + (r.y-self.me.y)**2;
-        if (r.team !== self.me.team
-            && SPECS.UNITS[self.me.unit].ATTACK_RADIUS[0] <= dist
-            && dist <= SPECS.UNITS[self.me.unit].ATTACK_RADIUS[1] ){
-            return true;
-        }
-        return false;
+		if (! self.isVisible(r)){
+			return false;
+		}
+		const dist = (r.x-self.me.x)**2 + (r.y-self.me.y)**2;
+		if (r.team !== self.me.team
+			&& SPECS.UNITS[self.me.unit].ATTACK_RADIUS[0] <= dist
+			&& dist <= SPECS.UNITS[self.me.unit].ATTACK_RADIUS[1] ){
+			return true;
+		}
+		return false;
 	});
-	
-	if (attackable.length>0){
+
+	if (attackable.length>0 && self.me.health > 50 ){
```

Comment:
Why do we care how much health the castle has? Shouldn't we just attack an enemy if it is close?

Reply:
yea you are right .. i will change this next sprint